src/rustdoc_gen.rs

```diff
@@ -110,6 +111,85 @@ impl<'a> CrateSource<'a> {
         all_crate_features.extend(self.regular_features());
         all_crate_features.into_iter().collect()
     }
+
+    /// Sometimes creates include features that are not meant for public use
+    /// or otherwise don't adhere to semver:
+    ///  - private features, like `bench` used for internal benchmarking only
+    ///  - nightly-only features, like `nightly`
+    ///  - unstable features containing experimental code, like `unstable`
+    ///
+    /// To ensure the best possible out=of-the-box user experience,
+    /// this function attempts to heuristically exclude feature names like above.
+    ///
+    /// The heuristics is based on the name since cargo does not currently include
+    /// a mechanism for marking features as private/hidden/unstable. When such
+    /// mechanisms are available in cargo, we'll update this functionality to make
+    /// use of them. Relevant cargo issues:
+    /// - unstable/nightly-only features: <https://github.com/rust-lang/cargo/issues/10881>
+    /// - private/hidden features:        <https://github.com/rust-lang/cargo/issues/10882>
+    ///
+    /// This function tries to filter mentioned features out.
+    /// Feature names not included by this function:
```

Comment:
This can also very easily be misinterpreted as the exact opposite of what it's trying to say, as if it's saying "this function tries to filter such features out, but explicitly does not do so for the following feature names: unstable, nightly, bench, ..."